### PR TITLE
fix: do not reconcile children that are iterable functions

### PR DESCRIPTION
**Pull request by raunofreiberg**
#11396.

Does this seem reasonable? Should we also add a warning?

### CODE REVIEW

File: packages/shared/ReactSymbols.js [hunk 1]
```diff
@@ -46,7 +46,11 @@ const MAYBE_ITERATOR_SYMBOL = typeof Symbol === 'function' && Symbol.iterator;
 const FAUX_ITERATOR_SYMBOL = '@@iterator';
 
 export function getIteratorFn(maybeIterable: ?any): ?() => ?Iterator<*> {
-  if (maybeIterable === null || typeof maybeIterable === 'undefined') {
+  if (
+    maybeIterable === null ||
+    typeof maybeIterable === 'undefined' ||
+    typeof maybeIterable === 'function'
```
Comment (gaearon) [inline packages/shared/ReactSymbols.js]: Maybe just `typeof maybeIterable !== 'object'`?

Comment (raunofreiberg) [reply packages/shared/ReactSymbols.js]: Makes sense.

File: packages/shared/ReactSymbols.js [hunk 2]
```diff
@@ -46,7 +46,11 @@ const MAYBE_ITERATOR_SYMBOL = typeof Symbol === 'function' && Symbol.iterator;
 const FAUX_ITERATOR_SYMBOL = '@@iterator';
 
 export function getIteratorFn(maybeIterable: ?any): ?() => ?Iterator<*> {
-  if (maybeIterable === null || typeof maybeIterable === 'undefined') {
+  if (
+    maybeIterable === null ||
+    typeof maybeIterable === 'undefined' ||
```
Comment (gaearon) [inline packages/shared/ReactSymbols.js]: You don't need this check now.

**Comment by gaearon** [comment]
Looks good. Thanks.

**Pull request merged**